installer/bundle/bundle_skel.sh

```diff
@@ -941,13 +941,21 @@ if [ "$installMode" = "R" -o "$installMode" = "P" ]; then
         fi
 
         pkg_rm omsconfig
-        pkg_rm omsagent
 
-        # If MDSD is installed and we're just removing (not purging), leave SCX
-        MDSD_INSTALLED=1
+        # If MDSD/LAD is installed and we're just removing (not purging), leave OMS, SCX and OMI
         check_if_program_exists_on_system azsec-mdsd
-        if [ $? -eq 0 -o -d /var/lib/waagent/Microsoft.OSTCExtensions.LinuxDiagnostic-*/mdsd ]; then
-            MDSD_INSTALLED=0
+        azsec_mdsd_installed=$?
+        check_if_program_exists_on_system lad-mdsd
+        lad_mdsd_installed=$?
+
+        # if at least one of mdsd product is installed
+        MDSD_INSTALLED=$(( $azsec_mdsd_installed && $lad_mdsd_installed ))
+
+        # If LAD was installed don't remove omsagent, but proceed if purge mode was selected
+        if [ $lad_mdsd_installed -ne 0 -o "$installMode" = "P"]; then
+            pkg_rm omsagent
```

Comment:
Why are we checking if installMode is Purge? In the case where LAD is installed on the VM and CX is running installMode as purge through shell bundle, This if statement will go and try to purge oms and still fail due to lad oms dependency issue right?

Reply:
That how it was done in the next line, I didn't want to change the behavior.

Reply:
Also our LAD/OMS extensions use "--remove", so it safe.

Reply:
But then in case where LAD is installed and then we uninstall LAD and it calls/tries to remove OMS. Wouldn't that just skip this pkg_rm omsagent since lad_mdsd_installed might return True (depending on at what step Lad uninstall process OMS removal is called) and not remove OMS even though we want to completely remove LAD and all it's dependencies?

Reply:
LAD will remove lad-mdsd first then calls omsagent-*.sh --remove next.
T